src/streamdiffusion/acceleration/tensorrt/engine_pool.py

```diff
@@ -126,11 +142,18 @@ def _compile_controlnet(self,
                            batch_size: int) -> bool:
         """Compile ControlNet to TensorRT"""
         try:
-            print(f"ControlNetEnginePool._compile_controlnet: Starting ControlNet compilation: {model_type} for {self.image_width}x{self.image_height}")
+            print(f"ControlNetEnginePool._compile_controlnet: Starting ControlNet compilation: {model_type} for dynamic 512-1024 range")
             
-            # Use actual image dimensions instead of hardcoded 512x512
-            height = self.image_height
-            width = self.image_width
+            # Use a flexible optimal resolution that allows for dynamic changes
+            # Instead of using current dimensions as optimal, use a middle value
+            # This allows the engine to handle both smaller and larger resolutions
+            min_resolution = 512
+            max_resolution = 1024
+            opt_resolution = (min_resolution + max_resolution) // 2  # 768x768 as optimal
```

Comment:
Ah found it! Is this the only thing we need to update it? Could we do this?
```suggestion
            min_resolution = 384
            max_resolution = 1024
            opt_resolution = (min_resolution + max_resolution) // 2  # 704x704 as optimal
```
(704 happened magically, exactly the value I wanted haha)

Reply:
I modified the min resolution to this value in a few places to work with our demo app as well.